knowledge-base-public/src/components/knowledge-base/KnowledgeBasePublicSection.js

```diff
@@ -8,12 +8,14 @@ import KnowledgeBasePublicNoAnswer from "./KnowledgeBasePublicNoAnswer";
 import React from "react";
 import { useRouter } from "next/router";
 import KnowledgeBaseSearch from "./KnowledgeBaseSearch";
-import { HiSearch } from "react-icons/hi";
+import { HiSearch, HiStar } from "react-icons/hi";
+import Link from "next/link";
 
 const KnowledgeBasePublicSection = ({ item, isRoot, isLoading, device }) => {
   const router = useRouter();
   const [sections, setSections] = useState(item?.children?.filter((c) => c.type === "section") || []);
   const [articles, setArticles] = useState(item?.children?.filter((c) => c.type === "article") || []);
+  const [topArticles, setTopArticle] = useState((item?.children?.filter((c) => c.type === "article") || []).sort((a, b) => b.read - a.read));
```

Comment:
Pas besoin de useState ici. topArticles est une variable dérivée d'un prop :
`const topArticles = (item?.children?.filter((c) => c.type === "article") || []).sort((a, b) => b.read - a.read);`
Elle est recalculée automatiquement à chaque fois que le prop "item" est modifié. Ilvaut mieux éviter de dupliquer le state sous peine de provoquer des comportements étranges (https://blog.bitsrc.io/duplication-in-state-management-a-story-of-a-bug-18c0d8206b59)